lib/HTFeed/Stage/Collate.pm

```diff
@@ -31,7 +31,8 @@ sub storages_from_config {
 
   my @storages;
   foreach my $storage_class (@{get_config('storage_classes')}) {
-    push(@storages, $storage_class->new(volume => $self->{volume}));
+    push(@storages, $storage_class->{class}->new(volume => $self->{volume},
```

Comment:
What about:

```perl
foreach my $storage_config (@{get_config('storage_classes')}) {
  push(@storages, HTFeed::Storage->from_config($storage_config);
}```

and add a method there to call the appropriate subclass constructor?

(There might also be a way to do this with just calling `HTFeed::Storage->new` and blessing it into the subclass, but I'd  rather avoid that as a) confusing and b) losing anything the subclass constructors might do)

Reply:
I would change `$storage_class` to `$storage_config` though